### PR TITLE
Some refactor to landing page tests to remove unused code

### PR DESCRIPTION
**Pull request by carlosfelix2**
## Summary
Some refactor to landing page tests to remove unused code

## Related issue(s)
https://app.zenhub.com/workspaces/mhv-on-vagov-top-level-view-62619a987d74510018ecc546/issues/gh/department-of-veterans-affairs/va.gov-team/72838

## Testing done
- e2e tests

## Screenshots
N/A - only the e2e tests were changed.

## What areas of the site does it impact?
- MHV Landing Page

## Acceptance criteria
- Assessed whether this test is necessary
- Rewrote test if deemed appropriate

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

### CODE REVIEW

File: src/applications/mhv-landing-page/tests/e2e/mhv-landing-page.cypress.spec.js [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
+      'featureToggles',
+    );
+  });
+
+  it('display the landing page when visiting root URL', () => {
+    cy.login(user);
+    cy.visit(rootUrl);
+    cy.injectAxeThenAxeCheck();
+    cy.findByRole('heading', { name: /^My HealtheVet$/i }).should.exist;
+    cy.findByRole('heading', { level: 2, name: /^Welcome/ }).should.exist;
```
Comment (carlosfelix2) [inline src/applications/mhv-landing-page/tests/e2e/mhv-landing-page.cypress.spec.js:17]: Note: Move the welcome message test here

Comment (radavis) [reply src/applications/mhv-landing-page/tests/e2e/mhv-landing-page.cypress.spec.js:19]: I think it makes sense to keep the tests for this component scoped to what is currently in production. In which case, the personalization feature flag should be off and `cy.findByRole('heading', { level: 2, name: /^Welcome/ }).should.not.exist;`.

Comment (radavis) [reply src/applications/mhv-landing-page/tests/e2e/mhv-landing-page.cypress.spec.js:21]: Yes. The file at `src/applications/mhv-landing-page/tests/e2e/personalization/welcome-message.cypress.spec.js` covers this.

File: src/applications/mhv-landing-page/tests/e2e/mhv-landing-page.cypress.spec.js [hunk 2]
```diff
@@ -1,9 +1,38 @@
...
+    ApiInitializer.initializeMessageData.withNoUnreadMessages();
+    cy.intercept('GET', '/v0/feature_toggles*', generateFeatureToggles()).as(
+      'featureToggles',
+    );
+    cy.login(user);
+    cy.visit(rootUrl);
     cy.injectAxeThenAxeCheck();
-    cy.findByRole('heading', { name: /we can’t find that page/ }).should.exist;
+    cy.findByRole('heading', { name: /^My HealtheVet$/i }).should.exist;
+    cy.findByRole('heading', { level: 2, name: /^Welcome/ }).should(
```
Comment (dcloud) [inline src/applications/mhv-landing-page/tests/e2e/mhv-landing-page.cypress.spec.js:34]: The test is named `'display the landing page and welcome message'`but the test looks for the absence of the welcome message. The name or assertion needs updating, I believe.

Also, is this redundant with the tests you added in `src/applications/mhv-landing-page/tests/e2e/personalization/welcome-message.cypress.spec.js`?

**Review by dcloud — COMMENTED** [review]
Seems good overall, though the test I commented on seems to have an inadvertent contradiction

File: src/applications/mhv-landing-page/tests/e2e/mhv-landing-page.cypress.spec.js (hunk 2 above)
Comment (carlosfelix2) [reply src/applications/mhv-landing-page/tests/e2e/mhv-landing-page.cypress.spec.js:34]: I changed the name

**Review by dcloud — APPROVED** [review]
Looks good, thanks for refactoring!

**Review by micahchiang — APPROVED** [review]

File: src/applications/mhv-landing-page/tests/e2e/personalization/welcome-message.cypress.spec.js [hunk 3]
```diff
@@ -7,20 +7,38 @@ import ApiInitializer from '../utilities/ApiInitializer';
...
+        mhvLandingPagePersonalization: true,
+      }),
+    ).as('featureToggles');
+    cy.login(user);
+    cy.visit(rootUrl);
+    cy.findByRole('heading', { level: 2, name: /^Welcome/ }).should.exist;
+    cy.injectAxeThenAxeCheck();
+  });
+
+  it('personalization enabled', () => {
```
Comment (radavis) [inline src/applications/mhv-landing-page/tests/e2e/personalization/welcome-message.cypress.spec.js:29]: This should read "personalization disabled".

**Review by radavis — APPROVED** [review]
Looks good. No blocking issues.

**Pull request merged**